test/int/wizards/breakpoints/implementation/internalFileBreakpointsWizard.ts

```diff
@@ -0,0 +1,84 @@
+import * as _ from 'lodash';
+import { ExtendedDebugClient } from 'vscode-chrome-debug-core-testsupport';
+import { findPositionOfTextInFile } from '../../../utils/findPositionOfTextInFile';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { PauseOnHitCount } from '../../../core-v2/chrome/internal/breakpoints/bpActionWhenHit';
+import { BreakpointWizard } from '../breakpointWizard';
+import { ValidatedMap } from '../../../core-v2/chrome/collections/validatedMap';
+import { FileBreakpointsWizard } from '../fileBreakpointsWizard';
+import { PromiseOrNot } from 'vscode-chrome-debug-core';
+import { BatchingUpdatesState } from './batchingUpdatesState';
+import { PerformChangesImmediatelyState } from './performChangesImmediatelyState';
+import { BreakpointsUpdater } from './breakpointsUpdater';
+import { BreakpointsWizard } from '../breakpointsWizard';
+
+export class BreakpointsUpdate {
+    public constructor(
+        public readonly toAdd: BreakpointWizard[],
+        public readonly toRemove: BreakpointWizard[],
+        public readonly toKeepAsIs: BreakpointWizard[]) { }
+}
+
+export interface IPerformChangesImmediatelyOrBatchState {
```

Comment:
`IPerformChangesImmediatelyOrBatchState` is really verbose. How about just `IBreakpointsState`.

Reply:
What do you think of IBatchOrNotState?

Reply:
In general I think of words like "or", "and", etc. when naming things as a code smell, as usually it's an indication that either we're doing too much with one object (violating single responsibility principle), or, as in this case, not clearly articulating what an object's responsibility _actually_ is.

`IBatchOrNotState` still tells me nothing about what the interface's responsibility actually is, only what it's possible values can be. That means that the only way I can truly understand what the interface is for is to go and look at the implementing classes.  The name of an interface should describe its purpose. What I see is that implementers of this interface do two things:

1) Define the strategy for breakpoint updates (batch vs individual)
2) Manage the current state of the breakpoints (e.g. which ones are set vs. not set)

Of those, it's the strategy that drives the behavior, and how we manage state, so I think an even better name would be something like `ISetBreakpointStrategy` or just `IBreakpointStrategy`, since we're using the strategy pattern to change the behavior of the system based on the implementation we choose.

Reply:
What do you think of IBreakpointsBatchingStrategy?

Reply:
I like it